app/code/Magento/Rule/Model/Condition/AbstractCondition.php

```diff
@@ -83,18 +83,14 @@ public function __construct(Context $context, array $data = [])
         $this->loadAttributeOptions()->loadOperatorOptions()->loadValueOptions();
 
         $options = $this->getAttributeOptions();
-        if ($options) {
-            foreach (array_keys($options) as $attr) {
-                $this->setAttribute($attr);
-                break;
-            }
+        if (!empty($options)) {
+            $attrs = array_keys($options);
+            $this->setAttribute($attrs[0]);
```

Comment:
More effective way to obtain key of the first element: https://stackoverflow.com/a/1028677/8640867